core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -447,6 +458,10 @@ public void delete(Delete delete) throws EngineException {
         } catch (OutOfMemoryError | IllegalStateException | IOException t) {
             maybeFailEngine("delete", t);
             throw new DeleteFailedEngineException(shardId, delete, t);
+        } finally {
+            if (delete.seqNo() != -1l) {
```

Comment:
[Field](https://github.com/elastic/elasticsearch/pull/14651/files#r45225330)? These are tricky to find right now. :)